File: investigation-backend/src/schemas/pomodoro.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, ConfigDict


class PomodoroCreate(BaseModel):
    task_id: UUID | None = None
    duration: int
    break_duration: int | None = None
    type: str | None = None
    status: str | None = None
    started_at: datetime | None = None


class PomodoroUpdate(BaseModel):
    duration: int | None = None
    break_duration: int | None = None
    type: str | None = None
    status: str | None = None
    started_at: datetime | None = None
    completed_at: datetime | None = None
    actual_duration: int | None = None


class PomodoroOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: UUID
    user_id: UUID
    task_id: UUID | None = None
    duration: int
    break_duration: int
    type: str
    status: str
    started_at: datetime
    completed_at: datetime | None = None
    actual_duration: int | None = None
    created_at: datetime

File: investigation-admin/src/api/pomodoro.ts
import { apiRequest } from "./client"

export type PomodoroSession = {
  id: string
  user_id: string
  task_id?: string | null
  duration: number
  break_duration: number
  type: string
  status: string
  started_at: string
  completed_at?: string | null
  actual_duration?: number | null
  created_at: string
}

export function fetchPomodoroSessions(params?: Record<string, any>): Promise<PomodoroSession[]> {
  return apiRequest<PomodoroSession[]>("/pomodoro/sessions/", { params })
}

export function createPomodoroSession(data: Record<string, any>): Promise<PomodoroSession> {
  return apiRequest<PomodoroSession>("/pomodoro/sessions/", {
    method: "POST",
    body: JSON.stringify(data)
  })
}

export function updatePomodoroSession(id: string, data: Record<string, any>): Promise<PomodoroSession> {
  return apiRequest<PomodoroSession>(`/pomodoro/sessions/${id}`, {
    method: "PUT",
    body: JSON.stringify(data)
  })
}

export function deletePomodoroSession(id: string): Promise<void> {
  return apiRequest<void>(`/pomodoro/sessions/${id}`, { method: "DELETE" })
}
